sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/CoalesceShufflePartitions.scala

```diff
@@ -33,6 +33,8 @@ import org.apache.spark.util.Utils
  */
 case class CoalesceShufflePartitions(session: SparkSession) extends AQEShuffleReadRule {
 
+  override def conf: SQLConf = session.sessionState.conf
```

Comment:
Can we change it to `override val conf: SQLConf = session.sessionState.conf` in cases where `conf` needs to be called multiple times?

Reply:
From the test results, there are other issues with changing to `val`. It seems we need to revert the last commit.

Reply:
Yes. I think so.

Reply:
sorry for misleading